ext/zlib/zlib.c

```diff
@@ -1157,14 +1157,29 @@ PHP_FUNCTION(deflate_add)
 	out_size = PHP_ZLIB_BUFFER_SIZE_GUESS(ctx->total_in + in_len);
 	out_size = (ctx->total_out >= out_size) ? 16 : (out_size - ctx->total_out);
 	out_size = (out_size < 16) ? 16 : out_size;
+	out_size += 64;
 	out = zend_string_alloc(out_size, 0);
 
 	ctx->next_in = (Bytef *) in_buf;
 	ctx->next_out = (Bytef *) ZSTR_VAL(out);
 	ctx->avail_in = in_len;
 	ctx->avail_out = ZSTR_LEN(out);
 
-	status = deflate(ctx, flush_type);
+	buffer_used = 0;
+
+	do {
+		if (ctx->avail_out == 0) {
+			/* more output buffer space needed; realloc and try again */
+			/* adding 64 more bytes solved every issue I have seen    */
+			/* the + 1 is for the string terminator added below */
+			out = zend_string_realloc(out, ZSTR_LEN(out) + 64 + 1, 0);
```

Comment:
overflow? Why? When an addition of 64 will overflow, we already will long have had an out of memory before...